magicgui/_qt.py

```diff
@@ -312,3 +340,106 @@ def setValue(self, value):
     def setMaximum(self, value):
         """Set maximum position of slider in float units."""
         super().setMaximum(value * self.PRECISION)
+
+
+class FileDialogMode(Enum):
+    """FileDialog mode options.
+
+    EXISTING_FILE - returns one existing file.
+    EXISTING_FILES - return one or more existing files.
+    OPTIONAL_FILE - return one file name that does not have to exist.
+    EXISTING_DIRECTORY - returns one existing directory.
+    R - read mode, returns one or more existing files.
+        Alias of EXISTING_FILES.
+    W - write mode, returns one file name that does not have to exist.
+        Alias of OPTIONAL_FILE.
+    """
+
+    EXISTING_FILE = "getOpenFileName"
+    EXISTING_FILES = "getOpenFileNames"
+    OPTIONAL_FILE = "getSaveFileName"
+    EXISTING_DIRECTORY = "getExistingDirectory"
+    # Aliases
+    R = "getOpenFileNames"
```

Comment:
```suggestion
    R = "getOpenFileName"
```

let's make this an alias for a single file.  and in a followup (or here if you want to) we can do something like this:
```python
@magicgui
def filepicker(filename=Sequence[Path("~")]):
    ...
```
and then in `type2widget` return `functools.partial(MagicFileDialog, mode='EXISTING_FILES')`

Reply:
whoops, sorry. I committed this and it broke the test.  can you fix that?  Then I think this is good to go

Reply:
Ok